### PR TITLE
:seedling: Included make targets for update binary

### PR DESCRIPTION
**Pull request by naveensrinivasan**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
feature

* **What is the current behavior?** (You can also link to an open issue here)
The update binary did not have make targets.

* **What is the new behavior (if this is a feature change)?**

  * Include the build and go mod verify targets to the update binary.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

None

* **Other information**:
None

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -116,6 +116,13 @@ dockerbuild: ## Runs docker build
...
+.PHONY: checkprojects
 checkprojects: ## Validates ./cron/projects.txt
 	cd ./scripts && go build  -o validate
 	./scripts/validate ./cron/projects.txt
+
+.PHONY: build-update
+build-update: ## builds the scripts/update binary
+	cd ./scripts/update && go mod tidy && go mod verify	
+	cd ./scripts/update && CGO=0 go build .
+	./scripts/tree-status
```
Comment (inferno-chromium) [inline Makefile:128]: build-update name is little confusing, this seems like projects-update ?
also, above checkprojects should be check-projects (dash in between).

File: Makefile [hunk 2]
```diff
@@ -116,6 +116,13 @@ dockerbuild: ## Runs docker build
...
 
+.PHONY: checkprojects
 checkprojects: ## Validates ./cron/projects.txt
 	cd ./scripts && go build  -o validate
 	./scripts/validate ./cron/projects.txt
+
+.PHONY: build-update
+build-update: ## builds the scripts/update binary
+	cd ./scripts/update && go mod tidy && go mod verify	
+	cd ./scripts/update && CGO=0 go build .
```
Comment (inferno-chromium) [inline Makefile:127]: Do you want "-a -ldflags '-w -extldflags "-static"'" ?

File: .gitignore [hunk 3]
```diff
@@ -3,6 +3,7 @@ scorecard
 gitblobcache
 validate
 scorecardcron
+scripts/update/update
```
Comment (inferno-chromium) [inline .gitignore:6]: Why update/update ?

**Review by azeemshaikh38 — APPROVED** [review]

File: .gitignore (hunk 3 above)
Comment (naveensrinivasan) [reply .gitignore]: Changed it to the `projects-update`

File: Makefile (hunk 2 above)
Comment (naveensrinivasan) [reply Makefile]: Agreed, thanks. Implemented it.

File: Makefile (hunk 1 above)
Comment (naveensrinivasan) [reply Makefile:128]: Agreed, took care of it. Thanks

**Pull request merged**